core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockDataMap.java

```diff
@@ -648,6 +652,51 @@ protected int getTotalBlocklets() {
     return sum;
   }
 
+  @Override public Map<String, Long> getRowCount(Segment segment, List<PartitionSpec> partitions,
+      Map<String, Long> blockletToRowCountMap) throws IOException {
+    if (taskSummaryDMStore.getTotalRowCount() == 0) {
+      return getRowCountForEachBlock(segment, partitions, blockletToRowCountMap);
+    } else {
+      if (taskSummaryDMStore.getRowCount() == 0) {
+        return new HashMap<>();
+      }
+      Long rowCount = blockletToRowCountMap.get("RowCount");
+      long totalRowCount = taskSummaryDMStore.getTotalRowCount();
+      if (null == rowCount) {
+        blockletToRowCountMap.put("RowCount", totalRowCount);
+      } else {
+        blockletToRowCountMap.put("RowCount", totalRowCount + rowCount);
```

Comment:
you key is always static `RowCount`. Any particular reason to use HashMap here? Can return type be changed to return the long value?

Reply:
HashMap is used to hold the map of block path vs the row count.
We have 2 cases:
(1) IUD table, in case of IUD table we need the actual block path from here so that it can be compared with the UpdateStatusManager later.
(2) Normal table(without IUD), where we don't need to do any comparisons later, so we are just putting only one entry to the Map with dummy key as "RowCount" and updating the value.